src/Data/Data.ts

```diff
@@ -2128,44 +2295,49 @@ export async function compareWithOldReceiptsData(lastStoredReceiptCycle = 0): Pr
 }
 
 export async function compareWithOldCyclesData(lastCycleCounter = 0): Promise<CompareResponse> {
-  const numberOfCyclesTocompare = 10
-  const start = lastCycleCounter - numberOfCyclesTocompare
-  const end = lastCycleCounter
-  const response = (await queryFromArchivers(
-    RequestDataType.CYCLE,
-    {
-      start,
-      end,
-    },
-    QUERY_TIMEOUT_MAX
-  )) as ArchiverCycleResponse
-  if (!response && !response.cycleInfo) {
-    throw Error(`Can't fetch data from cycle ${start} to cycle ${end}  from archivers`)
-  }
-  const downloadedCycles = response.cycleInfo
-  const oldCycles = await CycleDB.queryCycleRecordsBetween(start, end)
-  let success = false
-  let matchedCycle = 0
-  for (let i = 0; i < downloadedCycles.length; i++) {
-    // eslint-disable-next-line security/detect-object-injection
-    const downloadedCycle = downloadedCycles[i]
-    // eslint-disable-next-line security/detect-object-injection
-    const oldCycle = oldCycles[i]
-    if (
-      !downloadedCycle ||
-      !oldCycle ||
-      StringUtils.safeStringify(downloadedCycle) !== StringUtils.safeStringify(oldCycle)
-    ) {
-      console.log('Mismatched cycle Number', downloadedCycle.counter, oldCycle.counter)
-      return {
-        success,
-        matchedCycle,
+  try {
+    const numberOfCyclesTocompare = 10
+    const start = lastCycleCounter - numberOfCyclesTocompare
+    const end = lastCycleCounter
+    const response = (await queryFromArchivers(
+      RequestDataType.CYCLE,
+      {
+        start,
+        end,
+      },
+      QUERY_TIMEOUT_MAX
+    )) as ArchiverCycleResponse
+    if (!response && !response.cycleInfo) {
+      throw Error(`Can't fetch data from cycle ${start} to cycle ${end}  from archivers`)
+    }
+    const downloadedCycles = response.cycleInfo
+    const oldCycles = await CycleDB.queryCycleRecordsBetween(start, end)
+    let success = false
+    let matchedCycle = 0
+    for (let i = 0; i < downloadedCycles.length; i++) {
+      // eslint-disable-next-line security/detect-object-injection
+      const downloadedCycle = downloadedCycles[i]
+      // eslint-disable-next-line security/detect-object-injection
+      const oldCycle = oldCycles[i]
+      if (
+        !downloadedCycle ||
+        !oldCycle ||
+        StringUtils.safeStringify(downloadedCycle) !== StringUtils.safeStringify(oldCycle)
+      ) {
+        console.log('Mismatched cycle Number', downloadedCycle.counter, oldCycle.counter)
+        return {
+          success,
+          matchedCycle,
+        }
       }
+      success = true
+      matchedCycle = downloadedCycle.counter
     }
-    success = true
-    matchedCycle = downloadedCycle.counter
+    return { success, matchedCycle }
+  } catch (error) {
+    Logger.mainLogger.error(error)
```

Comment:
add some identifier text for log message
```suggestion
    Logger.mainLogger.error('compareWithOldCyclesData error: '+error)
```

Reply:
added